packages/flutter/lib/src/material/data_table_theme.dart

```diff
@@ -127,6 +155,8 @@ class DataTableThemeData with Diagnosticable {
       decoration: Decoration.lerp(a.decoration, b.decoration, t),
       dataRowColor: MaterialStateProperty.lerp<Color?>(a.dataRowColor, b.dataRowColor, t, Color.lerp),
       dataRowHeight: lerpDouble(a.dataRowHeight, b.dataRowHeight, t),
+      dataRowMinHeight: lerpDouble(a.dataRowMinHeight, b.dataRowMinHeight, t),
+      dataRowMaxHeight: lerpDouble(a.dataRowMaxHeight, b.dataRowMaxHeight, t),
```

Comment:
I think this is where the issue is. This is creating a value for all three, while the assertion only allows one or two to be set.